### PR TITLE
[docs] Improve v5.3.0 release notes

### PR DESCRIPTION
**Pull request by cherniavskii**
Fixes https://github.com/mui-org/material-ui-x/pull/3682#discussion_r790927168

I'll update the release page as well

### CODE REVIEW

**Comment by oliviertassinari** [comment]
Considering that we use the GitHub release page as the main entry point for the release (what we link in our outbound messages). Shouldn't we add a link to https://mui.com/blog/introducing-the-row-grouping-feature/ in the release page?

**Comment by cherniavskii** [comment]
> A different point. Considering that we use the GitHub release page as the main entry point for the release (what we link in our outbound messages, e.g. https://twitter.com/MUI_hq/status/1485637779874852869). Shouldn't we add a link to https://mui.com/blog/introducing-the-row-grouping-feature/ in the release page?

Sure, why not

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,13 +34,13 @@ A big thanks to the 9 contributors who made this release possible. Here are some
   />
   ```
 
-  See the [documentation](https://mui.com/components/data-grid/group-pivot/#row-grouping) for more details.
+  For more details see the [blog post](https://mui.com/blog/introducing-the-row-grouping-feature/) and [documentation](https://mui.com/components/data-grid/group-pivot/#row-grouping).
```
[the remark this answers is not in the document]
Comment (oliviertassinari) [reply CHANGELOG.md:37]:
```suggestion
  For more details see the [introduction blog post](https://mui.com/blog/introducing-the-row-grouping-feature/) and [documentation](https://mui.com/components/data-grid/group-pivot/#row-grouping).
```

**Review by oliviertassinari — APPROVED** [review]

**Comment by cherniavskii** [comment]
Updated release notes to reflect the changes

**Pull request merged**